### ISSUE TITLE
RA: engineers can not be placed in pillboxes

### ISSUE THREAD

**Issue by gdavegdave** [OpenRA/OpenRA#13449]
Engineers can not be placed in pillboxes; I think that's wrong. Every other infantry units can be put in pillboxes (even spies, dogs, zombies and ants!).

You can see an example of the impact of this here:
https://youtu.be/OiKjmCXjeTA?t=160
Hi tried to save his engi from the incoming ranger but was thwarted. I have confirmed that engis will not enter pbs in release-20170527.

Apparently, it used to be possible to put engis in pbs (according to Hi). Maybe it changed when the bug that allowed MCVs to be placed in camo pbs was fixed.

**Comment by obrakmann** [comment]
This is a bit of a pickle. If we make engis able to enter pillboxes, we'll at the same time introduce the bug where you can't use an engineer to repair one.

**Comment by pchote** [comment]
I see two options here:
1. (preferred, IMO): Give entering priority, and make the Force Attack modifier override it for repair.
2. Keep repair as priority, and make the force-move command prioritize entering.

**Comment by gdavegdave** [comment]
I agree @pchote  - your first option is preferable. How about having both Force Attack and Force Move effect the repair? That would be in line with the behaviour of the mechanic... I was in a game some days ago where @AMHOL tried to put a mechanic in a damaged phase transport and was rather miffed that he repaired it instead! Well, @anjew175 just tested now and found that Force Attack and Force Move does cause a mechanic to enter to a damaged phase.

**Comment by AMHOL** [comment]
I think I tried force moving it in there, but I have left click orders so some stuff doesn't work for me. You also can't use the "defend" stance with an engineer anymore and make it follow another unit.

**Comment by FrameLimiter** [comment]
Option #2 is how the Mechanic currently functions. Repair is the priority, but force-moving allows you to enter a damaged vehicle.(^as gdave/anjew have pointed out above)
I question how many players are even aware of this, and agree with GDave that option #1 would be the preferable option for both Engineers & Mechanics.

**Comment by Micr0Bit** [comment]
duplicate of #8128

**Comment by pchote** [comment]
> Introduced with commit ce6c883 in #7445
>
> Before that change, this bug existed in the reversed way: it was possible to enter, but engineers couldn't repair pillboxes. Just as broken.

**Comment by gdavegdave** [comment]
@atlimit8 this should be reopened as https://github.com/OpenRA/OpenRA/pull/17791 only allows an engineer to garrison a structure at 100% health. The preferred option 1 from https://github.com/OpenRA/OpenRA/issues/13449#issuecomment-306038669 hasn't been implemented.

**Comment by atlimit8** [comment]
Sorry, I forgot that it would be closed.
